Week5.java

```diff
@@ -0,0 +1,82 @@
+/*
+ Talk:
+ if T is null then a true is returned. (null is always a subtree)
+ If S is null then a false is returned.
+ Examples:
+ T is null , S is non empty : true
+ T is non empty, S is null : false
+ T and S are null : true
+ T is a similar to S subtree : true
+ T and S have 1 node and the data of both the nodes is equal : true
+ 
+ */
+
+package Homework;
+
+public class Week5 {
+	// tree T is similar to a subtree of S or not
+    public static boolean AreSame(TreeNode T, TreeNode S) {
+    	if(T==null && S==null) {
+    		return true;
+    	}
+    	if(T==null||S==null) {
+    		return false;
+    	}
+    	return (T.data==S.data && AreSame(T.left,S.left) && AreSame(T.right,S.right));
+      }
+    public static boolean AreSubtree(TreeNode T, TreeNode S) {
+    	if(T==null) {
+    		return true;
+    	}
+    	if(S==null) {
+    		return false;
+    	}
+    	if(AreSame(T,S)) {
+    		return true;
+    	}
+    	return (AreSubtree(T,S.left) || AreSubtree(T,S.right));
+    }
+    
+   
+	public static void main(String[] args) {
+		TreeNode S = new TreeNode(12);
+		/* Tree S

```

Comment:
Nice comments, much easier on a whiteboard in this case but easy to read so thank you :)